File: Leetcode_Ds_from_scratch/day4assignment/search2dMatrix.java
package Leetcode_Ds_from_scratch.day4assignment;


public class search2dMatrix {

    private static boolean matrixsearch(int[][]matrix , int target){
         for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[0].length;j++){
                if(matrix[i][j]==target) return true;
            }
        }
        return false;
    }

    
    public static void main(String[] args) {
        int[][]arr = {{1,2,3},{4,5,6},{7,8,9}};
        System.out.println(matrixsearch(arr, 9));
    }
}

File: 1_2026_PythonDsa/search2dMatrix.py
matrix = [[1,3,5,7],[10,11,16,20],[23,30,34,60]]
target=60


# binary search
newarr=[i for x in matrix for i in x ]
print(newarr)
def binarySearch(arr):
    
    low =0
    high=len(arr)
    

    while(low<high):
        mid =(low+high)//2
        if(arr[mid]==target):
            return mid
        if(arr[mid]<target):
            low=mid+1
        if(arr[mid]>target):
            high=mid-1
print(binarySearch(newarr))